src/main/java/demo/CreateIndexWorkflowStep.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package demo;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.flowframework.workflow.WorkflowData;
+import org.opensearch.flowframework.workflow.WorkflowStep;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Sample to show other devs how to pass data around.  Will be deleted once other PRs are merged.
+ */
+public class CreateIndexWorkflowStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateIndexWorkflowStep.class);
+
+    private final String name;
+
+    /**
+     * Instantiate this class.
+     */
+    public CreateIndexWorkflowStep() {
+        this.name = "CREATE_INDEX";
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+        CompletableFuture<WorkflowData> future = new CompletableFuture<>();
+        CompletableFuture.runAsync(() -> {
+            String inputIndex = null;
+            boolean first = true;
+            for (WorkflowData wfData : data) {
+                logger.debug(
+                    "{} sent params: {}, content: {}",
+                    first ? "Initialization" : "Previous step",
+                    wfData.getParams(),
```

Comment:
Are these parameters/content based on the REST API that each step is related to or are these the parameters/content of the fractal execute API which takes in the use case template and provisions the infrastructure?

Reply:
Yes and yes. :)

First note this is just an example class, @owaiskazi19 is working on actually implementing this in PR #38 

The `WorkflowData` class is a container for params (string to string map) and content (string to object map) which are intended to represent the two portions of a Rest Request you can communicate information (params on the URI, content in the content/data field).

The list that each step will get will have 1 or more of these instances:
 - index 0 comes from initialization, so may come from the template (user data) or the initial API call from a user
 - the remaining elements of the list come from the previous workflow step, where params will probably be empty (but isn't required to be) and the content will be the equivalent of the API response JSON.

Reply:
I see, thanks for the clarification. That PR #38 attempts to read the index name from the workflow data content rather than from the parameters. I've followed the same pattern (read everything from content) on my PR #44, but ill update this to read the pipelineId from parameters instead.